### PR TITLE
OC-21992 HTML code for double quotes appear in field history when text field is updated

### PR DESCRIPTION
**Pull request by theywa**
https://jira.openclinica.com/browse/OC-21992

#731

### CODE REVIEW

File: widget/discrepancy-note/dn-widget.js [hunk 1]
```diff
@@ -1599,6 +1599,13 @@ class Comment extends Widget {
         );
     }
 
+    _decodeHtml(str) {
+        if (str == null) {
+            return "";
+        }
+        return str.replaceAll("&lt;", "<").replaceAll("&gt;", ">").replaceAll("&#39;", "'").replaceAll("&quot;", "\"").replaceAll("&amp;", "&");
+    }
+
```
Comment (MartijnR) [inline widget/discrepancy-note/dn-widget.js:1608]: Did you compare the solution with `_encodeHtml` and chose string replacements with `replaceAll` for a reason? I am just curious. It is fine though.

**Review by MartijnR — APPROVED** [review]
I have the feeling there are more bugs to be discovered like this one. They are all caused by https://github.com/OpenClinica/enketo-express-oc/pull/722. However, I don't actually understand how that PR caused them (even when removing the quote replacements in `_encodeHtml` the bug still persists!

I think this solution is fine. However, if another bug like this pops up we'll need to start adding tests for all of them, and take a more holistic view of rendering text in that widget.

**Comment by MartijnR** [comment]
I restarted the CI testing because it may have been a fluke that it failed.

**Review by MartijnR — CHANGES_REQUESTED** [review]
Oops, looks like some eslint errors: https://github.com/OpenClinica/enketo-express-oc/actions/runs/7397885252/job/20209279340?pr=732#step:8:1744

File: widget/discrepancy-note/dn-widget.js (hunk 1 above)
Comment (theywa) [reply widget/discrepancy-note/dn-widget.js:1615]: No, I just think replaceAll is easier to use, but after seeing your commend I tried searching and it looks like replace is faster than replaceAll so I updated the code

**Comment by theywa** [comment]
> Oops, looks like some eslint errors: https://github.com/OpenClinica/enketo-express-oc/actions/runs/7397885252/job/20209279340?pr=732#step:8:1744

yes, I run eslint: fix now for easy fix

**Review by MartijnR — APPROVED** [review]

**Pull request merged**